src/elli_http.erl

```diff
@@ -583,6 +585,33 @@ do_check_max_size_x2(Socket, ContentLength, Buffer, MaxSize)
     elli_tcp:send(Socket, Response);
 do_check_max_size_x2(_, _, _, _) -> ok.
 
+%% @doc Same as check_max_size/5 but for arbitrary buffers (request line, headers)
+check_max_buffer_size(Socket, Buffer, Opts, {Mod, Args}, ErrorCode) ->
+    MaxSize = max_body_size(Opts),
+    BufferSize = size(Buffer),
+    case MaxSize > BufferSize of
+        true ->
+            ok;
+        false ->
+            case ErrorCode of
+                413 ->
+                    handle_event(Mod, bad_request, [{headers_size, BufferSize}], Args);
+                414 ->
+                    handle_event(Mod, bad_request, [{request_line_size, BufferSize}], Args)
+            end,
+
+            case MaxSize * 2 > BufferSize of
```

Comment:
Can't understand why a response is sent even for "way_too_big..." (in tests) requests